### PR TITLE
Mark required flags of clusterctl as required explicitly

### PR DESCRIPTION
**Pull request by ramineni**
This commit marks the flags -c, -m, -p and --provider
as required explicilty for `clusterctl create cluster`.
It also updates the corresponding help message to show
the flags are required for the command to run.

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note
'clusterctl create cluster ..' command now marks the 'cluster (-c), machines (-m), provider-components(-p), provider' flags as required explicitly. If any of the flags missed, command fails with error.
```

### CODE REVIEW

File: cmd/clusterctl/cmd/create_cluster.go [hunk 1]
```diff
@@ -118,12 +118,15 @@ func RunCreate(co *CreateOptions) error {
...
+	createClusterCmd.Flags().StringVarP(&co.Cluster, "cluster", "c", "", "A yaml file containing cluster object definition. Required.")
+	createClusterCmd.MarkFlagRequired("cluster")
+	createClusterCmd.Flags().StringVarP(&co.Machine, "machines", "m", "", "A yaml file containing machine object definition(s). Required.")
+	createClusterCmd.MarkFlagRequired("machines")
+	createClusterCmd.Flags().StringVarP(&co.ProviderComponents, "provider-components", "p", "", "A yaml file containing cluster api provider controllers and supporting objects. Required.")
+	createClusterCmd.MarkFlagRequired("provider-components")
 	// TODO: Remove as soon as code allows https://github.com/kubernetes-sigs/cluster-api/issues/157
-	createClusterCmd.Flags().StringVarP(&co.Provider, "provider", "", "", "Which provider deployment logic to use (google/vsphere/azure)")
-
+	createClusterCmd.Flags().StringVarP(&co.Provider, "provider", "", "", "Which provider deployment logic to use (google/vsphere/azure). Required.")
```
Comment (detiber) [inline cmd/clusterctl/cmd/create_cluster.go]: The test cases will need to be updated for the change in output, not sure about functionality changes related to marking as required.

**Comment by roberthbailey** [comment]
@ramineni - can you please update the initial pr description with a release note as well, since this changes the behavior of the cli?

**Comment by ramineni** [comment]
Thanks All for the comments, Ill update the patch accordingly.

File: cmd/clusterctl/cmd/create_cluster.go (hunk 1 above)
Comment (ramineni) [reply cmd/clusterctl/cmd/create_cluster.go]: done

**Pull request merged**